### ISSUE TITLE
Linux Build not displaying window

### ISSUE THREAD

**Issue by avarisclari** [GarageGames/Torque3D#1347]
I followed the instructions on the compile page to make the Engine, and when I try running the file it only outputs text in the cli with no window. I can't kill it with ctrl c either, and have to kill it from another interface.

```
Initializing platform...
Input Init:
% 
DebugDrawer Enabled!
GFX Init:
   Null device found
% 
Missing file: core/parseArgs.cs!
main.cs (87): Unable to find function defaultParseArgs
--------- Loading DIRS ---------
Missing file: levels/main.cs!
Error: Unable to find specified directory: levels
Missing file: art/main.cs!
Error: Unable to find specified directory: art
Missing file: scripts/main.cs!
Error: Unable to find specified directory: scripts
Missing file: tools/main.cs!
Error: Unable to find specified directory: tools
% 
--------- Parsing Arguments ---------
Engine initialized...
main.cs (261): Unable to find object: 'Canvas' attempting to call function 'showWindow'
Started up in 0.66 seconds...
% 
```

**Comment by eightyeight** [comment]
What's in the folder adjacent to the executable and `main.cs`? Sounds like you don't have any template files, but you're using `main.cs` from a template. Have a go following [these instructions](http://wiki.torque3d.org/introduction:creating-a-project-without-the-project-manager) and then see if it runs!

**Comment by avarisclari** [comment]
Thanks, that's an improvement, but now it segfaults. 
`````
Binding server port to default IP
UDP initialized on port 0
The requested renderer, D3D9, doesn't seem to be available. Trying the default, OpenGL.
The OpenGL renderer doesn't seem to be available. Trying the GFXNulDevice.
Attempting to create GFX device: GFX Null Device []
Initializing GFXCardProfiler (GFX Null Device Renderer)
   o Chipset : 'NULL Device'
   o Card    : 'GFX Null Device Card'
   o Version : '0'
   o VRAM    : 0 MB
   - Scanning card capabilities...
   - Loading card profiles...
      - No card profile core/profile/GFXNullDeviceRenderer.cs exists
      - No card profile core/profile/GFXNullDeviceRenderer.NULLDevice.cs exists
      - No card profile core/profile/GFXNullDeviceRenderer.NULLDevice.GFXNullDeviceCard.cs exists
      - No card profile core/profile/GFXNullDeviceRenderer.NULLDevice.GFXNullDeviceCard.0.cs exists
Device created, setting adapter and enumerating modes
% Segmentation fault (core dumped)
``````

**Comment by eightyeight** [comment]
Could you tell me about your system configuration? What OS, GPU, etc.?

**Comment by avarisclari** [comment]
Ubuntu 15.04
AMD Radeon HD 3650
Intel(R) Core(TM)2 Duo CPU     T9400  @ 2.53GHz
4 GB Ram

**Comment by hpvb** [comment]
I think this is fixed in the PR I have open at the moment. Could you test https://github.com/hpvb/Torque3D/tree/intel-mesa-compile-fix

**Comment by avarisclari** [comment]
It sort of is. I have a window now. Still having graphical issues though:

![torque](https://cloud.githubusercontent.com/assets/3046491/8614361/dfdce9ae-26b6-11e5-9ad1-4b7e0869aecc.png)

Here's the log too:
https://gist.github.com/avarisclari/8df2f58879deca533509

**Comment by hpvb** [comment]
Ah, I think I can fix this too! Just to be sure, could you please put the output of 'glxinfo' in a gist as well? 

I'll update the branch above when I have something for you to test.

**Comment by avarisclari** [comment]
Here you are. https://gist.github.com/avarisclari/7ddb576cda4df873dbdb
Always happy to help, I want a decent engine and I like TorqueScript. So any improvements to make it use able are awesome, even if all I can do really is report bugs. (I know some code, just not confident in it)

**Comment by eightyeight** [comment]
@hpvb thanks for jumping in here! I'l completely GL-illiterate so I was struggling :joy:

**Comment by hpvb** [comment]
Kind of a busy weekend, I'll probably get to this on Monday or Tuesday, sorry!

**Comment by hpvb** [comment]
Well, it was Monday, just a week later. Could you test the branch again please? I think that you should be able to get a somewhat usable editor. Make sure that the postfx are disabled, I haven't done much testing on that yet.

**Comment by avarisclari** [comment]
It works great now. Thanks. Moderators/owners/whoever should merge it :)

**Comment by hpvb** [comment]
I'm glad! We're still ironing out some issues with this and we may have to just completely replace GLEW but at least you can start work! :)

**Comment by eightyeight** [comment]
Great work @hpvb! Really appreciate your attention on this. Hopefully you and the others can get that issue fixed up and we can merge for 3.8.

**Comment by Areloch** [comment]
Getting #1567 should resolve all this once merged in.
